### PR TITLE
SI-10631 Use mongosh instead of MongoDB; upgrade Ubuntu to jammy-20231211.1

### PR DESCRIPTION
**Pull request by vkudelin**
# Resolves [SI-10631](https://suitespot.atlassian.net/browse/SI-10631)

**Changes purposed in this Pull request:**
- Update `ubuntu` from `jammy-20231004` to `jammy-20231211.1`
- Remove MongoDB
- Install `mongosh`

[SI-10631]: https://suitespot.atlassian.net/browse/SI-10631?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,26 +1,32 @@
...
-ENV PLATFORM="ubuntu1804"
+RUN apt-get install -y libcurl4 libgssapi-krb5-2 libldap-2.5-0 libwrap0 libsasl2-2 libsasl2-modules libsasl2-modules-gssapi-mit snmp openssl liblzma5
+ENV MONGO_VERSION="6.0.12" 
+ENV PLATFORM="ubuntu2204"
 ENV MONGO_DIR="mongodb-linux-x86_64-${PLATFORM}-${MONGO_VERSION}"
 ENV MONGO_TGZ="${MONGO_DIR}.tgz"
 RUN wget https://fastdl.mongodb.org/linux/${MONGO_TGZ} && \
   tar -zxvf ${MONGO_TGZ} && \
   rm ${MONGO_TGZ} && \
   cp ${MONGO_DIR}/bin/* /usr/bin
```
Comment (wfairclough) [inline Dockerfile]: Do we still need the entire Mongo installed in this container?  I thought we only use `mongosh` now.

**Review by wfairclough — COMMENTED** [review]
1 question

**Review by wfairclough — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (vkudelin) [reply Dockerfile]: good q. I even didn't think about that

Comment (vkudelin) [reply Dockerfile]: I have tested and didn't see any problem with excluding MongoDB from the image

**Pull request merged**